org.osgi.test.junit5.cm/src/test/java/org/osgi/test/junit5/cm/ConfigAnnotationTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.osgi.test.junit5.cm;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.test.common.annotation.config.ConfigEntry;
+import org.osgi.test.common.annotation.config.InjectConfiguration;
+import org.osgi.test.common.annotation.config.WithConfiguration;
+import org.osgi.test.common.annotation.config.WithFactoryConfiguration;
+
+@ExtendWith(ConfigurationExtension.class)
+@WithConfiguration(pid = ConfigAnnotationTest.NONSTATIC_CONFIGURATION_PID)
+public class ConfigAnnotationTest {
+
+	public static final String							FACTORY_CONFIGURATION_PID	= "my.factory.configuration.pid";
+	public static final String							NONSTATIC_CONFIGURATION_PID	= "nonstatic.configuration.pid";
+
+	private static ServiceReference<ConfigurationAdmin>	sref						= null;
+	private static BundleContext						bundleContext				= null;
+	private static ConfigurationAdmin					ca							= null;
+
+	@BeforeAll
+	public static void beforeAll() {
+
+		bundleContext = FrameworkUtil.getBundle(ConfigAnnotationTest.class)
+			.getBundleContext();
+
+		sref = bundleContext.getServiceReference(ConfigurationAdmin.class);
+		if (sref == null) {
+			throw new IllegalStateException("The configuration admin service cannot be found.");
+		}
+
+		ca = bundleContext.getService(sref);
```

Comment:
We should be use ServiceExtention to get services for testing :-) Eat our own food.

Reply:
done